packages/core/data-transfer/src/engine/index.ts

```diff
@@ -916,10 +942,15 @@ class TransferEngine<
     const destination = await this.destinationProvider.createAssetsWriteStream?.();
 
     const transform = this.#createStageTransformStream(stage);
-    const tracker = this.#progressTracker(stage, {
-      size: (value: IAsset) => value.stats.size,
-      key: (value: IAsset) => extname(value.filename) || 'No extension',
-    });
+    // Use the perChunk tracker for assets
+    const tracker = this.#progressTracker(
```

Comment:
Instead of modifying the generic method, we could also just have a dedicated method to create an assetProgressTracker? Might be easier to maintain

Reply:
Yeah, I can split it off into a different method instead of using an option